halo/attest/keeper/keeper_internal_test.go

```diff
@@ -0,0 +1,317 @@
+package keeper
+
+import (
+	"testing"
+
+	"github.com/omni-network/omni/halo/attest/types"
+
+	k1 "github.com/cometbft/cometbft/crypto/secp256k1"
+	cmttypes "github.com/cometbft/cometbft/types"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	fuzz "github.com/google/gofuzz"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeeper_Add(t *testing.T) {
+	t.Parallel()
+
+	// hard-coded test data
+	var (
+		blockHashes []common.Hash
+		vals        = []k1.PrivKey{k1.GenPrivKey(), k1.GenPrivKey(), k1.GenPrivKey()}
+		val1        = cmttypes.NewValidator(vals[0].PubKey(), 10)
+		val2        = cmttypes.NewValidator(vals[1].PubKey(), 15)
+		val3        = cmttypes.NewValidator(vals[2].PubKey(), 15)
+		blockRoot   = []byte{11, 5, 3, 67, 25, 34}
+	)
+	fuzz.New().NilChance(0).NumElements(3, 3).Fuzz(&blockHashes)
+
+	// cmp transformation options to ignore private fields of proto generated types.
+	var (
+		atteTrans = cmp.Options{cmpopts.IgnoreUnexported(Attestation{})}
+		sigsTrans = cmp.Options{cmpopts.IgnoreUnexported(Signature{})}
```

Comment:
can probably combine this into a single `cmpOpts`. Also meaning of "*Trans" suffix isn't clear?

Reply:
The trans is short for transformer(s) since before the comparison we might want to transform the data to make sure that the comparison is always predictable (e.g. sorting arrays before comparing them etc.). I just followed the examples in the library. But I will rename it.